client.go

```diff
@@ -42,6 +43,14 @@ type Client interface {
 	// offset, OffsetNewest for the offset of the message that will be produced next, or a time.
 	GetOffset(topic string, partitionID int32, time int64) (int64, error)
 
+	// Coordinator returns the coordinating broker for a consumer group. It will return a locally cached
+	// value if it's available. This may be stale, in which case you should call RefreshCoordinator
+	// to update the locally cached value.
+	Coordinator(consumerGroup string) (*Broker, error)
+
+	// RefreshCoordinator refreshes the coordinator for a consumer group.
```

Comment:
OK, I feel like it's just over-complicated to return the result, and *then* write it to the cache in a separate method, when you could simply write it to the cache in the first place. `getCoordinator`'s result is never going to be used for anything but updating the cache, so splitting that apart seems over-engineered to me.

We agree on `Coordinator`/`Leader`/etc should just read from the cache. I'm just not sure why we need two separate methods (fetch from remote, update cache) when that feels like a single operation to me.